File: 2025Irrgarten/src/Irrgarten/Weapon.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package Irrgarten;

/**
 *
 * @author astralisx
 */
public class Weapon {
    private float power;
    private int uses;
    
    public Weapon(float power,int uses){
        this.power=power;
        this.uses=uses;
    }
    
    public float attack(){
        if (this.uses > 0){
            uses--;
            return this.power;
        }
        return 0;
    }

    public boolean discard(){
        return Dice.discardElement(this.uses);
    }

    @Override
    public String toString(){
        return String.format("W[%f,%d]",power,uses);
    }
}

File: IrrgartenRuby/weapon.rb
# encoding: UTF-8

require_relative 'dice'

module Irrgarten
  class Weapon
    def initialize(power, uses)
      @power = power.to_f
      @uses = uses.to_i
    end

    def attack
      if @uses > 0
        @uses -= 1
        return @power
      else
        return 0.0
      end
    end

    def discard
      return Dice.dicard_element(@uses)
    end

    def to_s
      return "W[#{@power}, #{@uses}]"
    end
  end
end
